flink-yarn/src/main/java/org/apache/flink/yarn/cli/FlinkYarnSessionCli.java

```diff
@@ -530,6 +530,11 @@ protected Configuration applyCommandLineOptionsToConfiguration(CommandLine comma
 	private boolean isYarnPropertiesFileMode(CommandLine commandLine) {
 		boolean canApplyYarnProperties = !commandLine.hasOption(addressOption.getOpt());
 
+		// if canApplyYarnProperties is false,return quickly !
+		if (!canApplyYarnProperties) {
+			return canApplyYarnProperties;
```

Comment:
Maybe simplify to `return false` or even just wrap the loop in `if (canApplyYarnProperties) { }` so that there is just a single returning point.

Reply:
+1 to the latter

Reply:
done!